src/gc/impl/manual/gc.d

```diff
@@ -0,0 +1,305 @@
+
+    void* calloc(size_t size, uint bits, const TypeInfo ti) nothrow
+    {
+        void* p = cstdlib.calloc(1, size);
+
+        if (size && p is null)
+            onOutOfMemoryError();
+        return p;
+    }
+
+    void* realloc(void* p, size_t size, uint bits, const TypeInfo ti) nothrow
+    {
+        p = cstdlib.realloc(p, size);
+
+        if (size && p is null)
+            onOutOfMemoryError();
+        return p;
+    }
+
+    size_t extend(void* p, size_t minsize, size_t maxsize, const TypeInfo ti) nothrow
+    {
+        return 0;
+    }
+
+    size_t reserve(size_t size) nothrow
+    {
+        return 0;
+    }
+
+    void free(void* p) nothrow
+    {
+        cstdlib.free(p);
+    }
+
+    /**
+     * Determine the base address of the block containing p.  If p is not a gc
+     * allocated pointer, return null.
+     */
+    void* addrOf(void* p) nothrow
+    {
+        return null;
+    }
+
+    /**
+     * Determine the allocated size of pointer p.  If p is an interior pointer
+     * or not a gc allocated pointer, return 0.
+     */
+    size_t sizeOf(void* p) nothrow
+    {
+        return 0;
+    }
+
+    /**
+     * Determine the base address of the block containing p.  If p is not a gc
+     * allocated pointer, return null.
+     */
+    BlkInfo query(void* p) nothrow
+    {
+        return BlkInfo.init;
+    }
+
+    GCStats stats() nothrow
+    {
+        return GCStats.init;
+    }
+
+    void addRoot(void* p) nothrow @nogc
+    {
+        Root* r = cast(Root*) cstdlib.realloc(roots, (nroots + 1) * roots[0].sizeof);
+        if (r is null)
+            onOutOfMemoryError();
+        r[nroots++] = p;
+        roots = r;
+    }
+
+    void removeRoot(void* p) nothrow @nogc
+    {
+        for (size_t i = 0; i < nroots; ++i)
+        {
+            if (roots[i] is p)
+            {
+                roots[i] = roots[--nroots];
+                return;
+            }
+        }
+        assert(false);
+    }
+
+    @property RootIterator rootIter() @nogc
+    {
+        return &rootsApply;
+    }
+
+    private int rootsApply(scope int delegate(ref Root) nothrow dg)
+    {
+        int result = 0;
+        for (int i = 0; i < nroots; i++)
+        {
+            result = dg(roots[i]);
+
+            if (result)
+                break;
+        }
+
+        return result;
+    }
+
+    void addRange(void* p, size_t sz, const TypeInfo ti = null) nothrow @nogc
+    {
+        Range* r = cast(Range*) cstdlib.realloc(ranges, (nranges + 1) * ranges[0].sizeof);
+        if (r is null)
+            onOutOfMemoryError();
+        r[nranges].pbot = p;
+        r[nranges].ptop = p + sz;
+        r[nranges].ti = cast() ti;
+        ranges = r;
+        ++nranges;
+    }
+
+    void removeRange(void* p) nothrow @nogc
+    {
+        for (size_t i = 0; i < nranges; ++i)
+        {
+            if (ranges[i].pbot is p)
+            {
+                ranges[i] = ranges[--nranges];
+                return;
+            }
+        }
```

Comment:
Same suggestion as for `removeRoot` - verify correctness and preferably use a container.